packages/core/src/components/axes/toolbar.ts

```diff
@@ -379,47 +431,81 @@ export class Toolbar extends Component {
 	}
 
 	getControlConfigs() {
-		const numberOfIcons = Tools.getProperty(
-			this.getOptions(),
-			'toolbar',
-			'numberOfIcons'
-		);
+		const numberOfIcons =
+			Tools.getProperty(this.getOptions(), 'toolbar', 'numberOfIcons') -
+			1;
 		const controls = Tools.getProperty(
 			this.getOptions(),
 			'toolbar',
 			'controls'
 		);
 
-		const controlList = [];
 		const overflowSpecificControls = [];
+		const buttonList = [];
+		const overflowList = [];
+
+		let displayIcons = true;
+
 		controls.forEach((control) => {
-			const controlConfig = this.getControlConfigByType(control.type);
+			let controlConfig = null;
+			// check if button is custom or default control
+			if (control.type === ToolbarControlTypes.CUSTOM) {
+				// add generic id if missing
+				if (Tools.getProperty(control, 'id') === null) {
+					// add id directly to the data passed so that id isn't reassigned on rerender
+					control.id = `toolbar-button-${Toolbar.buttonID}`;
+					Toolbar.buttonID++;
+				}
+				// define function if missing
+				if (Tools.getProperty(control, 'shouldBeDisabled') === null) {
+					control.shouldBeDisabled = () => false;
+				}
+
+				controlConfig = control;
+			} else {
+				controlConfig = this.getControlConfigByType(control.type);
+			}
 
 			// add to list if config is valid
 			if (controlConfig) {
 				controlConfig.text = control.text ? control.text : control.type;
 
 				if (controlConfig.id.indexOf('toolbar-export') !== -1) {
 					overflowSpecificControls.push(controlConfig);
+				} else if (buttonList.length < numberOfIcons && displayIcons) {
+					// check if icon exists
+					// if button does not exist, all of the preceding items will be added to overflow menu
+					// this will ensure users use correct order & fill in any missing data
+					if (
+						Tools.getProperty(
+							controlConfig,
+							'iconSVG',
+							'content'
+						) === null
+					) {
+						displayIcons = false;
```

Comment:
hmmm... why do this? why wouldn't we just move anything we find with no icons into the overflow menu?

Reply:
this seems like an undocumented or unexpected side-effect